File: Mathematical Algorithms/Mersenne Prime/Mersenne.py
# Generate all prime numbers less than n.
def SieveOfEratosthenes(n, prime):
    for i in range(0, n + 1):
        prime[i] = True
    p = 2
    while p * p <= n:
        if prime[p] == True:
            for i in range(p * 2, n + 1, p):
                prime[i] = False
        p += 1


# Function to generate mersenne primes less than or equal to n
def mersennePrimes(n):
    # Create a boolean array "prime[0..n]"
    prime = [0] * (n + 1)
    # Generating primes using Sieve
    SieveOfEratosthenes(n, prime)
    # Generate all numbers of the form 2^k - 1 and smaller than or equal to n.
    k = 2
    while ((1 << k) - 1) <= n:
        num = (1 << k) - 1
        # Checking whether number is prime and is one less then the power of 2
        if prime[num]:
            print(num, end=" ")
        k += 1

File: Mathematical Algorithms/Mersenne Prime/Mersenne.cpp
#include <iostream>
using namespace std;

// Generate all prime numbers less than n.
void SieveOfEratosthenes(int n, bool prime[]) {
    for (int i = 0; i <= n; i++)
        prime[i] = true;
    for (int p = 2; p * p <= n; p++) {
        if (prime[p] == true) {
            for (int i = p * 2; i <= n; i += p)
                prime[i] = false;
        }
    }
}

// Function to generate mersenne primes less than or equal to n
void mersennePrimes(int n) {
    bool prime[n + 1];
    SieveOfEratosthenes(n, prime);
    for (int k = 2; ((1 << k) - 1) <= n; k++) {
        long long num = (1 << k) - 1;
        if (prime[num])
            cout << num << " ";
    }
}